plugins/catalog-backend/src/ingestion/LocationAnalyzer.ts

```diff
@@ -37,12 +37,16 @@ export class RepoLocationAnalyzer implements LocationAnalyzer {
       kind: 'Component',
       metadata: {
         name: name,
-        // Probably won't handle properly self-hosted git providers with custom url
-        annotations: { [`${source}/project-slug`]: `${owner}/${name}` },
       },
       spec: { type: 'other', lifecycle: 'unknown' },
     };
 
+    if (source === 'github.com') {
+      entity.metadata.annotations = {
+        [`github.com/project-slug`]: `${owner}/${name}`,
```

Comment:
Let's also add other public Git host providers i.e. `gitlab.com`, `bitbucket.org` here, since those will be missed by this change.

Reply:
@OrkoHunter I wasn't aware those were valid annotations? They're not currently mentioned in the [docs](https://backstage.io/docs/features/software-catalog/well-known-annotations). Is there plugins that use those annotations?

Looking at the `AnnotateScmSlugEntityProcessor`, it seems to also only support `github.com/project-slug`:
https://github.com/backstage/backstage/blob/cbfa75bf69da90961daae517f28a106327ceb5d0/plugins/catalog-backend/src/ingestion/processors/AnnotateScmSlugEntityProcessor.ts#L26

Reply:
@kuangp Hmm. I noticed that the existing code was setting these annotations anyway, which will now be removed by this change.

```ts
        annotations: { [`${source}/project-slug`]: `${owner}/${name}` },
```

This `source` here can totally be `gitlab.com` and `bitbucket.org`. Hence, my suggestion would be to not disable setting these annotations here.

But I agree they might not be in use yet.

Reply:
Wouldn't it make more sense to just rely on the AnnotateScmSlugEntityProcessor entirely rather then handle GitHub specially here?

Reply:
`AnnotateScmSlugEntityProcessor` is an optional processor and not added by default so I kept this to preserve existing behavior. I'm fine also with removing this annotation here, the bug in #3541 was preventing us from using the catalog file PR feature since it was adding an invalid annotation for us.

Reply:
@OrkoHunter updated my change to also whitelist `gitlab.com` and `bitbucket.org` to preserve current behavior for these SCMs.